### DISCUSSION TITLE
How to run an action (here: "deploying" through sftp) only when needed?

### DISCUSSION THREAD

**Discussion by I-is-as-I-does** [community/community#25978] [Actions]
<p>Hi everyone!</p>
<p><em>(TL;DR at the bottom)</em></p>
<p>I’m new to git and gitHub; sorry if my issue sounds newbiesque!<br>
Here is the situation:</p>
<ul>
<li>
<p>my repo has 2 basic branches: <strong>dev</strong>, and <strong>prod</strong></p>
</li>
<li>
<p>I’ve setup a <strong>pre-push hook</strong> for “dev”, useful when I’m working on my computer: I ask winSCP (a ftp client) to synch files with my dev webserver (low cost hosting plan :flushed:).<br>
It’s basic automation, that allows me to run/test my code (mostly php) in the same conditions than prod (because my local wamp isn’t always reliable) + for other non-coder people working on the project to give me feedbacks on new features.</p>
</li>
<li>
<p>To achieve the same thing when I’m on the go and making edits on my phone, I’ve setup an <strong>action</strong> on GitHub : <a href="https://github.com/SamKirkland/FTP-Deploy-Action" rel="nofollow noopener">FTP Deploy</a>.<br>
The job takes at least 1mn, but hey, that’s ok.<br>
I’ve created a new branch: <strong>dev_golive</strong>, and this action runs <em>on: push: branches: dev_golive</em>.</p>
</li>
</ul>
<p>Now, my issue is this:</p>
<ul>
<li>how to automatically keep “dev_golive” up-to-date with the “dev” branch? Auto-merging?.. Auto-PR?.. Auto “double-push”?..</li>
<li>OR, could I delete this branch, and only run the GitHub action when a commit message contains a certain keyword? Maybe that would trigger a tag, or a label, or something that can be recognized as an action event? :thinking:
</li>
<li>or?..</li>
</ul>
<p>I’ve looked for actions that can achieve something like this on GitHub marketplace, googled for tutorials, with no avail…<br>
What am I missing? Is there a more obvious way to proceed? :thinking:</p>
<p><strong>TL;DR</strong>: The goal is to run an action (in this case: “deploying” through sftp) only when needed. How to run this action on a custom event? Or, if using a specific branch, how to automatically keep that branch in synch with the “main” one?</p>
<p>Thanks a lot. :pray:</p>

**Comment by I-is-as-I-does** [comment, marked as answer]
<p>Well. Indeed, it was VERY simple. Hours spent finding this… :disappointed:<br>
I just needed to add this line in the yml file, before the action:</p>
<p><code>if: "contains(toJSON(github.event.commits.*.message), '[somekeyword]')"</code></p>
<p>Hope that will help someone else.</p>

**Comment by facundofarias** [comment]
I know you are trying to use a Github Action, but maybe you can give [DeployHQ](https://www.deployhq.com/) a try!
